runtime/ms-rest/lib/serialization.js

```diff
@@ -484,6 +484,46 @@ function deserializeDictionaryType(mapper, responseBody, objectName) {
   return responseBody;
 }
 
+/**
+ * Get the value of the property with the provided path. This function will first do a
+ * case-sensitive check for the property, and if that returns undefined then it will do a
+ * case-insensitive check.
+ * @param {object} value The value to get the property value from.
+ * @param {string[]} propertyPath The path to the property.
+ * @returns {any} The value of the property or undefined if it wasn't found.
+ */
+function getPropertyValue(value, propertyPath) {
```

Comment:
I would recommend writing some unit tests for this scenario to make sure that everything is covered.

Property present with null value
Property present with different casing and the value is null
Property present with non null value
Property present with different casing with a non-null value

Test the above scenarios with flattening involved.

Reply:
I added these unit tests in this PR: https://github.com/Azure/azure-sdk-for-node/pull/4147